torch/utils/data/_utils/pin_memory.py

```diff
@@ -46,7 +46,9 @@ def _pin_memory_loop(in_queue, out_queue, device_id, done_event, device):
 
 
 def pin_memory(data, device=None):
-    if isinstance(data, torch.Tensor):
+    if hasattr(data, "pin_memory"):
+        return data.pin_memory()
+    elif isinstance(data, torch.Tensor):
```

Comment:
is this equivalent to a more colloquial `if torch.is_tensor(data)`?

Reply:
Looks like it, but the documentation indicates that it's preferred to use `isinstance(data, torch.Tensor)` for explicitness:

> Note that this function is simply doing isinstance(obj, Tensor). Using that isinstance check is better for typechecking with mypy, and more explicit - so it’s recommended to use that instead of is_tensor.